doc/usage/execution/applications.rst

```diff
@@ -0,0 +1,126 @@
+.. Copyright Celeritas contributors: see top-level COPYRIGHT file for details
+.. SPDX-License-Identifier: CC-BY-4.0
+
+.. highlight:: none
+
+Applications
```

Comment:
we might want to eventually port this to a manpage

Reply:
Yeah, I'd like to start using CLI11 at some point, and maybe auto-generate the docs from that.